ruby/rails/security/injection/tainted-sql-string.yaml

```diff
@@ -1,54 +1,61 @@
-  message: Detected user input used to manually construct a SQL string. This is usually
-    bad practice because manual construction could accidentally result in a SQL injection.
-    An attacker could use a SQL injection to steal or modify contents of the database.
-    Instead, use a parameterized query which is available by default in most database
-    engines. Alternatively, consider using an object-relational mapper (ORM) such
-    as ActiveRecord which will protect your queries.
+  message: Detected user input used to manually construct a SQL string. This is
+    usually bad practice because manual construction could accidentally result
+    in a SQL injection. An attacker could use a SQL injection to steal or
+    modify contents of the database. Instead, use a parameterized query which
+    is available by default in most database engines. Alternatively, consider
+    using an object-relational mapper (ORM) such as ActiveRecord which will
+    protect your queries.
   metadata:
-    cwe: 'CWE-89: Improper Neutralization of Special Elements used in an SQL Command
-      (''SQL Injection'')
-
-      '
+    cwe: >
+      CWE-89: Improper Neutralization of Special Elements used in an SQL
+      Command ('SQL Injection')
     owasp:
-    - A01:2017 - Injection
-    - A03:2021 - Injection
+      - A01:2017 - Injection
+      - A03:2021 - Injection
     category: security
     technology:
-    - rails
+      - rails
     references:
-    - https://rorsecurity.info/portfolio/ruby-on-rails-sql-injection-cheat-sheet
+      - https://rorsecurity.info/portfolio/ruby-on-rails-sql-injection-cheat-sheet
     license: Commons Clause License Condition v1.0[LGPL-2.1-only]
   mode: taint
   pattern-sources:
-  - patterns:
-    - pattern-either:
-      - pattern: params
-      - pattern: request
+    - patterns:
+        - pattern-either:
+            - pattern: params
+            - pattern: request
   pattern-sinks:
-  - patterns:
-    - pattern-either:
-      - patterns:
-        - pattern: |
-            "...#{$EXPR}..."
-        - pattern-not-inside: |
-            $FUNC("...",  "...#{$EXPR}...",...)
-        - focus-metavariable: $EXPR
-        - pattern-regex: (?i)(select|delete|insert|create|update|alter|drop)\b|\w+\s*!?[<>=].*
-      - patterns:
+    - patterns:
         - pattern-either:
-          - pattern: Kernel::sprintf("$SQLSTR", $EXPR)
-          - pattern: |
-              "$SQLSTR" + $EXPR
-          - pattern: |
-              "$SQLSTR" % $EXPR
-        - pattern-not-inside: |
-            $FUNC("...",  "...#{$EXPR}...",...)
-        - focus-metavariable: $EXPR
-        - metavariable-regex:
-            metavariable: $SQLSTR
-            regex: (?i)(select|delete|insert|create|update|alter|drop)\b
+            - patterns:
+                - pattern-either:
+                    - pattern: |
+                        $RECORD.where($X,...)
+                    - pattern: |
+                        $RECORD.find(..., :conditions => $X,...)
```

Comment:
should this be => ...,$X,...

Reply:
the sink is the `conditions` string - it lets you specify a partial raw SQL query. `$Y.find(...)` is SELECT FROM WHERE <conditions> under the hood: `User.find(:first, :conditions => "user_id = '#{params[:user][:user_id]}'")`